packages/venia-ui/lib/components/ProductFullDetail/productFullDetail.js

```diff
@@ -98,14 +99,25 @@ const ProductFullDetail = props => {
     const [optionCodes] = useState(derivedOptionCodes);
 
     // Members.
-    const { amount: productPrice } = product.price.regularPrice;
     const classes = mergeClasses(defaultClasses, props.classes);
     const isMissingOptions = getIsMissingOptions(product, optionSelections);
     const mediaGalleryEntries = getMediaGalleryEntries(
         product,
         optionCodes,
         optionSelections
     );
+    
+    const isConfigurable = isProductConfigurable(product);
+    
+    let productPrice = product.price.regularPrice
+    if (isConfigurable) {
+                const item = findMatchingVariant({
```

Comment:
We already do this once in [getMediaGalleryEntries](https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/talons/ProductFullDetail/useProductFullDetail.js#L79). Can you find a way to refactor so that we only have to do this once?

Reply:
yes sure Thanks for your comments @sirugh . 
I am working on changes will try to update soon